tests/unit/log_capture/test_console_log_capture.py

```diff
@@ -178,15 +207,20 @@ def test_console_log_capture_thread_captures_both_stdout_and_stderr(no_capture):
     thread.join()
 
     # then
-    logs_sink.assert_any_call({"monitoring/stdout": "Hello stdout"}, 1, ANY)
-    logs_sink.assert_any_call({"monitoring/stderr": "Hello stderr"}, 1, ANY)
-    logs_sink.assert_any_call({"monitoring/stdout": "World stdout"}, 2, ANY)
-    logs_sink.assert_any_call({"monitoring/stderr": "World stderr"}, 2, ANY)
+    logs_sink.assert_any_call({"monitoring/stdout": "Hello stdout"}, _make_step(1), ANY)
+    logs_sink.assert_any_call({"monitoring/stderr": "Hello stderr"}, _make_step(1), ANY)
+    logs_sink.assert_any_call({"monitoring/stdout": "World stdout"}, _make_step(2), ANY)
+    logs_sink.assert_any_call({"monitoring/stderr": "World stderr"}, _make_step(2), ANY)
 
 
 LINE_LIMIT = 1024 * 1024
 
 
+def _make_step(index: int, initial: float = 0) -> float:
```

Comment:
Please don't use prod code in tests. I know it's tested in another test, but still.